exporters/prometheus/src/exporter_utils.cc

```diff
@@ -167,6 +175,159 @@ std::string PrometheusExporterUtils::SanitizeNames(std::string name)
   return name;
 }
 
+std::regex INVALID_CHARACTERS_PATTERN("[^a-zA-Z0-9]");
+std::regex CHARACTERS_BETWEEN_BRACES_PATTERN("\\{(.*?)\\}");
+std::regex SANITIZE_LEADING_UNDERSCORES("^_+");
+std::regex SANITIZE_TRAILING_UNDERSCORES("_+$");
+std::regex SANITIZE_CONSECUTIVE_UNDERSCORES("[_]{2,}");
+
+std::string PrometheusExporterUtils::GetEquivalentPrometheusUnit(
+    const std::string &raw_metric_unit_name)
+{
+  if (raw_metric_unit_name.empty())
+  {
+    return raw_metric_unit_name;
+  }
+
+  std::string converted_metric_unit_name = RemoveUnitPortionInBraces(raw_metric_unit_name);
+  converted_metric_unit_name = ConvertRateExpressedToPrometheusUnit(converted_metric_unit_name);
+
+  return CleanUpString(GetPrometheusUnit(converted_metric_unit_name));
+}
+
+std::string PrometheusExporterUtils::GetPrometheusUnit(const std::string &unit_abbreviation)
+{
+  static std::map<std::string, std::string> units{// Time
```

Comment:
Is is better to use unordered_map and string_view here?

Reply:
thanks, done.

Reply:
What I mean is use hash map to find unit conversion here and below, not just the return type of `TranslateToPrometheus`.

string_view may not be necessary here, because all string value is small. In some old implementation it will use COW and the new implementation will use small string optimization to reduce malloc.

Reply:
@owent could you please check my last commit?